exec/java-exec/src/main/java/org/apache/drill/exec/store/pcapng/PcapngFormatPlugin.java

```diff
@@ -47,7 +47,7 @@ public PcapngFormatPlugin(String name, DrillbitContext context, Configuration fs
 
   public PcapngFormatPlugin(String name, DrillbitContext context, Configuration fsConf, StoragePluginConfig config, PcapngFormatConfig formatPluginConfig) {
     super(name, context, fsConf, config, formatPluginConfig, true,
-        false, true, false,
+        false, true, true,
```

Comment:
Isn't the middle `true` wrong? It is for `blockSplittable`. That means we'll start reading at an arbitrary block boundary. Since this is a binary format, it is not clear that we can scan forward to the beginning of the next record as can be done in Sequence File and (restricted) CSV.

Also, if the file is zip-encoded, then it is never block splittable since Zip files cannot be read at an arbitrary offset.

This creates an issue: the block-splittable attribute right now is a constant. But, if any file is zip-encoded, then it is never block splittable. Any way to handle this fact?

And, any way to test this behaviour?

Reply:
1. Drill uses `BlockMapBuilder` to split file into blocks if possible. According to its code, it tries to split the file if `blockSplittable` is set to true and file IS NOT compressed. So even if format is block splittable but came as compressed file, it won't be split.

https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/store/schedule/BlockMapBuilder.java#L115

Looks like most of compressed formats are not splittable, that's why Drill considers any compressed file not splittable: https://i.stack.imgur.com/jpprr.jpg

2. Regarding blockSplittable for Pcang format, you are right such format is not splittable, as well as Pcap, I have updated the value of `blockSplittable` to `false` for both formats.

https://blog.marouni.fr/pcap2seq/